sycl/include/CL/sycl/stl.hpp

```diff
@@ -22,34 +22,52 @@
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
- template < class T, class Alloc = std::allocator<T> >
- using vector_class = std::vector<T, Alloc>;
+#if defined(_WIN32) && !defined(__SYCL_DEVICE_ONLY__)
+namespace detail {
+// SYCL library is designed such a way that STL objects cross DLL boundary,
+// which is not guaranteed to work and considered not safe in general.
+// Only using same dynamic C++ runtime library for sycl[d].dll and for
+// the application using sycl[d].dll is guaranteed to work properly.
+inline constexpr bool isMSVCDynamicCXXRuntime() {
```

Comment:
`constexpr` implies `inline`

```suggestion
constexpr bool isMSVCDynamicCXXRuntime() {
```